batch/batch/google_storage.py

```diff
@@ -84,10 +89,16 @@ def _delete_gs_files(self, uri_prefix):
         bucket, prefix = GCS._parse_uri(uri_prefix)
         bucket = self.gcs_client.bucket(bucket)
         for blob in bucket.list_blobs(prefix=prefix):
-            blob.delete()
+            try:
+                blob.delete()
+            except google.api_core.exceptions.NotFound:
+                continue
 
     def _delete_gs_file(self, uri):
         bucket, path = GCS._parse_uri(uri)
         bucket = self.gcs_client.bucket(bucket)
         f = bucket.blob(path)
```

Comment:
This should be inside the try as well. Presumably this could fail if the path does not exist.

Reply:
I don't think so. I think it's just initializing an object in Python. Doesn't actually do any API requests. https://googleapis.dev/python/storage/latest/buckets.html#google.cloud.storage.bucket.Bucket.blob

Reply:
great